### ISSUE TITLE
Add option for full batch learning with different sized validation set

### ISSUE THREAD

**Issue by tnolle** [lmjohns3/downhill#10]
As you suggested, an option like `batch_size=0` or `batch_size=-1` could force the trainer to do full batch on both the training and the validation set without creating a list index out of bounds error.

**Comment by lmjohns3** [comment]
Fixed this with ba5f6f1.

Will be released in a few minutes as downhill v0.3.1 -- please upgrade with ``pip install -U downhill``.
